src/views/Home.jsx

```diff
@@ -1,26 +1,61 @@
 import './Home.css';
+import { useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import { generateToken } from '@the-collab-lab/shopping-list-utils';
+import { getItemData, streamListItems } from '../api';
+
+export function Home({ setListToken }) {
+	const [errorMessage, setErrorMessage] = useState('');
+	const [joinListToken, setJoinListToken] = useState('');
 
-export function Home({ makeNewList }) {
 	const navigate = useNavigate();
 
-	// use callback function so the newToken that's generated in this child component
-	// can be passed up to the parent component (App.js)
-	function handleClick() {
+	function makeNewList() {
 		const newToken = generateToken();
-		makeNewList(newToken);
+		setListToken(newToken);
 		navigate('/list');
 	}
 
+	function joinListSubmit(event) {
+		event.preventDefault();
+
+		return streamListItems(joinListToken, (snapshot) => {
+			const data = getItemData(snapshot);
+			if (data.length) {
+				setListToken(joinListToken);
+				navigate('/list');
+			} else {
+				setErrorMessage('Could not join the list.');
+			}
+		});
+	}
+
+	function joinListChange(event) {
+		if (errorMessage) setErrorMessage('');
+		setJoinListToken(event.target.value);
+	}
 	return (
 		<div className="Home">
 			<p>
 				Hello from the home (<code>/</code>) page!
 			</p>
-			<button type="button" onClick={handleClick}>
+			<button type="button" onClick={makeNewList}>
 				Make new list
 			</button>
+			<form onSubmit={joinListSubmit}>
+				<label htmlFor="share-list">
+					Join a List
+					<input
+						id="share-list"
+						pattern="(?:\w+ ){2}\w+"
+						title="Token must be three words separated with spaces."
+						onChange={joinListChange}
+						value={joinListToken}
+					></input>
+				</label>
+				<button type="submit">Submit</button>
+			</form>
+			<p>{errorMessage}</p>
```

Comment:
I would recommend adding a conditional here so that there is not an empty paragraph tag when there is no error message:

```suggestion
			{errorMessage.length > 0 && <p>{errorMessage}</p>}
```

Reply:
ty @luisaugusto ! Adjusted!